skbio/io/util.py

```diff
@@ -37,17 +39,53 @@ def _is_string_or_bytes(s):
     return isinstance(s, str) or isinstance(s, bytes)
 
 
-def _get_filehandle(filepath_or, *args, **kwargs):
+def get_filehandle(filepath_or, *args, **kwargs):
     """Open file if `filepath_or` looks like a string/unicode/bytes, else
     pass through.
     """
     if _is_string_or_bytes(filepath_or):
-        fh, own_fh = open(filepath_or, *args, **kwargs), True
+        if filepath_or.endswith('.gz'):
```

Comment:
There isn't a strong guarantee of a filename actually existing. Additionally it would be better if it checked for the [magic number](http://en.wikipedia.org/wiki/Gzip#File_format) (`1f 8b`) of a gzip file.